### PR TITLE
ddl: change drop partition and truncate partition's job args to support multi partition id array

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: AilinKid <314806019@qq.com>

### What problem does this PR solve?

it is prepared for cancel `add multi partition action` which is in mid state 

Problem Summary: change drop partition and truncate partition's job args to support multi partition id array

### What is changed and how it works?

What's Changed: 
1: change drop partition and truncate partition's job args to multi id array
2: support ddl worker to handle these two action and add ids to delete range table.

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Breaking backward compatibility

### Release note 

- change drop partition and truncate partition's job args to support multi partition id array
- Notice: This is a in-compatible change and we can not:
1: rolling upgrade the cluster during the drop partition DDL operation
2: degrade to old tidb if the model data has been written and the DDL operation is not finished

### CODE REVIEW

File: ddl/partition.go [hunk 1]
```diff
@@ -576,38 +576,48 @@ func validRangePartitionType(col *model.ColumnInfo) bool {
...
 	}
 	oldDefs := pi.Definitions
-	for _, def := range oldDefs {
-		if strings.EqualFold(def.Name.L, strings.ToLower(partName)) {
-			if len(oldDefs) == 1 {
-				return errors.Trace(ErrDropLastPartition)
+	for _, pn := range partNames {
+		found := false
+		for _, def := range oldDefs {
+			if strings.EqualFold(def.Name.L, strings.ToLower(pn)) {
```
Comment (kennytm) [inline ddl/partition.go:588]: ok it's not introduced by this PR, why do we do `EqualFold(L, ToLower(pn))` 😱 either the EqualFold or the ToLower should be removed

```suggestion
			if def.Name.L == strings.ToLower(pn) {
```

File: ddl/partition.go [hunk 2]
```diff
@@ -576,38 +576,48 @@ func validRangePartitionType(col *model.ColumnInfo) bool {
...
 	oldDefs := tblInfo.Partition.Definitions
 	newDefs := make([]model.PartitionDefinition, 0, len(oldDefs)-1)
-	var pid int64
-	for i := 0; i < len(oldDefs); i++ {
-		if !strings.EqualFold(oldDefs[i].Name.L, strings.ToLower(partName)) {
-			continue
+	var pids []int64
+	for _, partName := range partNames {
+		for i := 0; i < len(oldDefs); i++ {
+			if !strings.EqualFold(oldDefs[i].Name.L, strings.ToLower(partName)) {
```
Comment (kennytm) [inline ddl/partition.go:610]:
```suggestion
			if oldDefs[i].Name.L != strings.ToLower(partName) {
```

**Comment by tiancaiamao** [comment]
LGTM
This is a in-compatible change and we can not:
1. rolling upgrade the cluster during the drop partition DDL operation
2. degrade to old tidb if the model data has been written and the DDL operation is not finished

File: ddl/partition.go [hunk 3]
```diff
@@ -576,38 +576,48 @@ func validRangePartitionType(col *model.ColumnInfo) bool {
...
 	}
 	oldDefs := pi.Definitions
-	for _, def := range oldDefs {
-		if strings.EqualFold(def.Name.L, strings.ToLower(partName)) {
-			if len(oldDefs) == 1 {
-				return errors.Trace(ErrDropLastPartition)
+	for _, pn := range partNames {
+		found := false
+		for _, def := range oldDefs {
+			if def.Name.L == strings.ToLower(pn) {
```
Comment (zimulala) [inline ddl/partition.go:588]: `pn` is lower(https://github.com/pingcap/tidb/pull/18419/files#diff-edfa54cbe2fdb418f0a77bd0ec4a56f2R2812), so we needn't do `ToLower`.

Comment (AilinKid) [reply ddl/partition.go:588]: good

File: ddl/partition.go (hunk 2 above)
Comment (AilinKid) [reply ddl/partition.go]: good, addressed PTAL

**Pull request merged**